tests/functionaltests/Tests/ToastNotificationTests.mm

```diff
@@ -125,92 +125,117 @@ - (void)application:(UIApplication*)application didReceiveToastAction:(NSDiction
-}
+    TEST_CLASS_CLEANUP(ToastNotificationForegroundTestClassCleanup) {
+        return FunctionalTestCleanupUIApplication();
+    }
 
-TEST(ToastNotificationTest, ActivatedAppReceivesToastNotification) {
-    LOG_INFO("Activated App Receives Toast Notification Test: ");
-
-    ActivatedAppReceivesToastNotificationDelegate* testDelegate = [ActivatedAppReceivesToastNotificationDelegate new];
-    [[UIApplication sharedApplication] setDelegate:testDelegate];
-
-    // Create mocked data to pass into application
-    RTObject* value = [WFPropertyValue createString:@"TOAST_NOTIFICATION_TEST_VALUE"];
-    WFCValueSet* wfcUserInput = [WFCValueSet make];
-    [wfcUserInput setObject:value forKey:@"TOAST_NOTIFICATION_TEST_KEY"];
-
-    auto fakeToastNotificationActivatedEventArgs = Make<MockToastNotificationActivatedEventArgs>();
-    fakeToastNotificationActivatedEventArgs->Setget_Argument([](HSTRING* argument) {
-        Wrappers::HString value;
-        value.Set(L"TOAST_NOTIFICATION_TEST");
-        *argument = value.Detach();
-        return S_OK;
-    });
-
-    fakeToastNotificationActivatedEventArgs->Setget_UserInput([&wfcUserInput](IPropertySet** userInput) {
-        wfcUserInput.comObj.CopyTo(userInput);
-        return S_OK;
-    });
-
-    fakeToastNotificationActivatedEventArgs->Setget_Kind([](ActivationKind* kind) {
-        *kind = ActivationKind_ToastNotification;
-        return S_OK;
-    });
-
-    fakeToastNotificationActivatedEventArgs->Setget_PreviousExecutionState([](ApplicationExecutionState* state) {
-        *state = ApplicationExecutionState_Running;
-        return S_OK;
-    });
-
-    // Calls OnActivated, which should not go through activation because we are activated
-    // But should still call our new delegate method
-    UIApplicationActivationTest(reinterpret_cast<IInspectable*>(fakeToastNotificationActivatedEventArgs.Get()),
-                                NSStringFromClass([ToastNotificationForegroundActivationTestDelegate class]));
-
-    NSDictionary* methodsCalled = [testDelegate methodsCalled];
-    EXPECT_TRUE(methodsCalled);
-    EXPECT_TRUE([methodsCalled objectForKey:@"application:didReceiveToastAction:"]);
-}
+    TEST_METHOD(ForegroundActivationDelegateMethodsCalled) {
+        ToastNotificationForegroundActivationTestDelegate* testDelegate = [[UIApplication sharedApplication] delegate];
+        NSDictionary* methodsCalled = [testDelegate methodsCalled];
+        EXPECT_TRUE(methodsCalled);
+        EXPECT_TRUE([methodsCalled objectForKey:@"application:willFinishLaunchingWithOptions:"]);
+        EXPECT_TRUE([methodsCalled objectForKey:@"application:didFinishLaunchingWithOptions:"]);
+        EXPECT_TRUE([methodsCalled objectForKey:@"application:didReceiveToastAction:"]);
+    }
+};
+
+class ActivatedAppReceivesToastNotificationTests {
+    BEGIN_TEST_CLASS(ActivatedAppReceivesToastNotificationTests)
+    END_TEST_CLASS()
```

Comment:
I know but I'm not changing how it was for people already. This is just transplanted code. #lazy.